// ngx-mat-timepicker/esm2022/lib/pipes/ngx-mat-timepicker-minutes-formatter.pipe.mjs
import { Pipe } from "@angular/core";
import { NgxMatTimepickerUtils } from "../utils/ngx-mat-timepicker.utils";
import * as i0 from "@angular/core";
export class NgxMatTimepickerMinutesFormatterPipe {
    transform(minute, gap = NgxMatTimepickerUtils.DEFAULT_MINUTES_GAP) {
        if (!minute) {
            return minute;
        }
        return minute % gap === 0 ? minute : "";
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.2.4", ngImport: i0, type: NgxMatTimepickerMinutesFormatterPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "17.2.4", ngImport: i0, type: NgxMatTimepickerMinutesFormatterPipe, isStandalone: true, name: "minutesFormatter" }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.2.4", ngImport: i0, type: NgxMatTimepickerMinutesFormatterPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: "minutesFormatter",
                    standalone: true
                }]
        }] });
